### PR TITLE
Adjust auto tuning to be on by default and only allow override if hig…

### PR DESCRIPTION
**Pull request by mikebell90**
…her.

### CODE REVIEW

File: client/src/main/java/com/opentable/jaxrs/CalculateThreads.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class CalculateThreads {
+    private static final Logger LOG = LoggerFactory.getLogger(CalculateThreads.class);
+
+    private CalculateThreads() {
+        /* utility class */
+    }
+    public static int calculateThreads(final int executorThreads) {
```
Comment (adamzr) [inline client/src/main/java/com/opentable/jaxrs/CalculateThreads.java]: Add JavaDoc

Comment (adamzr) [inline client/src/main/java/com/opentable/jaxrs/CalculateThreads.java]: Suggest changing `executorThread` to "configuredNumberOfExecutorThreads`

Comment (adamzr) [inline client/src/main/java/com/opentable/jaxrs/CalculateThreads.java]: Suggest changing method name to `calculateMaximumNumberOfExecutorThreads`

File: client/src/main/java/com/opentable/jaxrs/CalculateThreads.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        // For current standard 8 core machines this is 10 regardless.
+        // On Java 10, you might get less than 8 core reported, but it will still size as if it's 8
+        // Beyond 8 core you MIGHT undersize if running on Docker, but otherwise should be fine.
+        final int optimalThreads= Math.max(10, (Runtime.getRuntime().availableProcessors() + 2 ));
+        int threadsChosen = optimalThreads;
+        if (executorThreads > optimalThreads) {
+            threadsChosen = executorThreads;
```
Comment (adamzr) [inline client/src/main/java/com/opentable/jaxrs/CalculateThreads.java]: Add comment explaining why

File: client/src/main/java/com/opentable/jaxrs/CalculateThreads.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        }
+
```
Comment (adamzr) [inline client/src/main/java/com/opentable/jaxrs/CalculateThreads.java]: If threadsChosen != executorThreads, i.e. we are ignoring the configured thread count, I think we should log a warning.

File: client/src/main/java/com/opentable/jaxrs/CalculateThreads.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    public static int calculateThreads(final int executorThreads, final String name) {
+        // For current standard 8 core machines this is 10 regardless.
+        // On Java 10, you might get less than 8 core reported, but it will still size as if it's 8
+        // Beyond 8 core you MIGHT undersize if running on Docker, but otherwise should be fine.
+        final int optimalThreads= Math.max(BASE_OPTIMAL_THREADS, (Runtime.getRuntime().availableProcessors() + THREAD_OVERHEAD));
+        int threadsChosen = optimalThreads;
+        if (executorThreads > optimalThreads) {
+            // They requested more, sure we can do that!
+            threadsChosen = executorThreads;
+            LOG.debug("Requested more than optimal threads. This is not necessarily an error, but you may be overallocating.");
```
Comment (rushalias) [inline client/src/main/java/com/opentable/jaxrs/CalculateThreads.java]: warning message is better here in my opinion.

File: client/src/main/java/com/opentable/jaxrs/CalculateThreads.java (hunk 1 above)
Comment (mikebell90) [reply client/src/main/java/com/opentable/jaxrs/CalculateThreads.java]: I thought this was too twee ;)

File: client/src/main/java/com/opentable/jaxrs/CalculateThreads.java (hunk 2 above)
Comment (mikebell90) [reply client/src/main/java/com/opentable/jaxrs/CalculateThreads.java]: comment added.

File: client/src/main/java/com/opentable/jaxrs/CalculateThreads.java (hunk 3 above)
Comment (mikebell90) [reply client/src/main/java/com/opentable/jaxrs/CalculateThreads.java]: done

**Pull request merged**